client/landing/stepper/declarative-flow/site-setup-flow.ts

```diff
@@ -184,7 +185,14 @@ const siteSetupFlow: Flow = {
 						pendingActions.push( saveSiteSettings( siteId, { launchpad_screen: 'full' } ) );
 					}
 
-					Promise.all( pendingActions ).then( () => window.location.assign( to ) );
+					let redirectionUrl = to;
+
+					// Forcing cache invalidation to retrieve latest launchpad_screen option value
+					if ( isLaunchpadIntent( intent ) ) {
+						redirectionUrl = addQueryArgs( { postOnboarding: true }, to );
```

Comment:
if we are only adding the arg for the launchpad flows, maybe a `showLaunchpad` or similar verbage makes more sense than `postOnboarding` ?